### PR TITLE
Feature/customer orders food

### PR DESCRIPTION
**Pull request by solskinIsak**
This pull request includes significant enhancements to the `Ordering.API` project, focusing on improving the order management and Kafka integration functionalities. Key changes include the addition of new DTOs, controllers, and services, as well as modifications to the database schema and GitHub Actions workflow.

### Enhancements to Order Management:

* **Order Controller Addition:**
  * Added `OrderController` to handle order-related API endpoints, including retrieving orders by ID. (`Ordering.API/Controllers/OrderController.cs`)
  * Removed the obsolete `WeatherForecastController`. (`Ordering.API/Controllers/WeatherForecastController.cs`)

* **Database Schema Updates:**
  * Updated `Order` and `Orderline` tables to use UUIDs with default values generated by `uuid_generate_v4()`. (`Database/database-script.sql`)
  * Added `CREATE EXTENSION IF NOT EXISTS "uuid-ossp";` for UUID generation. (`Database/database-script.sql`)

* **DTOs and Validators:**
  * Introduced new DTOs for handling order and cart data (`CartDto`, `CartItemDto`, `DishDto`, `OrderDto`, `OrderLineDto`). (`Ordering.API/RequestDTOs/CartDto.cs`, `Ordering.API/RequestDTOs/CartItemDto.cs`, `Ordering.API/RequestDTOs/DishDto.cs`, `Ordering.API/ResponseDTOs/OrderDto.cs`, `Ordering.API/ResponseDTOs/OrderLineDto.cs`) [[1]](diffhunk://#diff-f324c6ee72b6136d7a15c44f914afcf1556e82615bf5b0184962740e3d64b3bdR1-R21) [[2]](diffhunk://#diff-ae5c2521175ad0f466f7e2b0667f2385635da6543105f6084cf910fa3d8adb5cR1-R9) [[3]](diffhunk://#diff-17ab5caa31ac3f73f95c5f722fe59684d370bf6cd0d5d3d229c942c6e4487b44R1-R8) [[4]](diffhunk://#diff-a56565c717b28476b948e8e9d963bdf142f48b1c8b888bce47e15875841243c5R1-R38) [[5]](diffhunk://#diff-f0684e932184c483affb2624e088973e6886fa751af3706a4d6197d4c2f439adR1-R25)
  * Added FluentValidation for `CartDto`. (`Ordering.API/Validators/CartRequestValidator.cs`)

### Kafka Integration Enhancements:

* **Kafka Consumer Updates:**
  * Enhanced `KafkaConsumer` to deserialize messages into `CartDto`, validate them, and create orders using `OrderService`. (`Ordering.API/Kafka/KafkaConsumer.cs`) [[1]](diffhunk://#diff-22623d3585dc5af06f910c7131c47215cff9d45b182c56fd57739583dc0bcb09L1-R39) [[2]](diffhunk://#diff-22623d3585dc5af06f910c7131c47215cff9d45b182c56fd57739583dc0bcb09L46-R76)

* **Kafka Producer Service:**
  * Added `KafkaProducerService` to produce order messages to Kafka. (`Ordering.API/Services/KafkaProducerService.cs`, `Ordering.API/Services/IKafkaProducerService.cs`) [[1]](diffhunk://#diff-08f375bf826c079de9f81fb24bd9396b42abe1f466d188a46eb6454304c05498R1-R9) [[2]](diffhunk://#diff-7a2eab1cd81e225a33f696cbbc283535e0bdadda0cdc26a42e2d713b7010916aR1-R20)

### GitHub Actions Workflow:

* **Workflow Enhancements:**
  * Updated `test-runner.yml` to include release triggers and added steps for building, testing, and reporting code coverage. (`.github/workflows/test-runner.yml`)

### Miscellaneous:

* **Project File Updates:**
  * Added new package references for FluentValidation, EntityFrameworkCore, and other dependencies. (`Ordering.API/Ordering.API.csproj`)
  * Updated HTTP request examples for new order-related endpoints. (`Ordering.API/Ordering.API.http`)

These changes collectively enhance the functionality, reliability, and maintainability of the `Ordering.API` project, particularly in the areas of order management and Kafka messaging.

### CODE REVIEW

**Review by HelenaLykstoft — COMMENTED** [review]
LGTM, godt med tests der dækker det meste

**Review by thejamiegc — CHANGES_REQUESTED** [review]
Jeg ville nok gerne se du laver noget refactor på din controller. Og så kigget på test igen måske kig på hvordan vi andre har gjort dem.

**Review by HelenaLykstoft — COMMENTED** [review]
LGTM, men måske lige enten indkommenter eller slet de tests der er udkommenteret :D 

 - det stod i mine commits at man sku ignorere de udkommenterede tests x) -Isak

File: Ordering.IntegrationTests/OrderTests.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/*using Microsoft.AspNetCore.Mvc;
```
Comment (thejamiegc) [inline Ordering.IntegrationTests/OrderTests.cs:1]: maybe delete this?

File: Ordering.UnitTests/Services/KafkaProducerServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        var kafkaProducerService = new KafkaProducerService(kafkaProducer.Object);
+        
+        kafkaProducer.Setup(x => x.ProduceAsync(It.IsAny<string>(),It.IsAny<string>(), It.IsAny<Order>()))
+            .Returns(Task.CompletedTask);
+
+        // Act
+        await kafkaProducerService.ProduceOrderAsync(order);
+
+        // Assert
+        kafkaProducer.Verify(x => x.ProduceAsync("order.accepted", $"{order.CustomerUserName}", It.Is<Order>(o => o.CustomerUserName == "Ikit Kugelblitz")), Times.Once);
```
Comment (thejamiegc) [inline Ordering.UnitTests/Services/KafkaProducerServiceTest.cs:44]:
```suggestion
        kafkaProducer.Verify(x => x.ProduceAsync("order.accepted", order.CustomerUserName, It.Is<Order>(o => o.CustomerUserName == "Ikit Kugelblitz")), Times.Once);
```

File: qodana.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#-------------------------------------------------------------------------------#
```
Comment (thejamiegc) [inline qodana.yaml:1]: skal denne file med?

**Review by thejamiegc — CHANGES_REQUESTED** [review]
All most there kid you got some good looking now. but still some minor shit in there. so try again. 

ps. maybe just think the first time then i don't review such god awful code. ❤️

**Review by thejamiegc — APPROVED** [review]
fine i will approve then

**Pull request merged**